toolchain/check/testdata/interop/cpp/function/operators.carbon

```diff
@@ -784,6 +784,80 @@ fn F() {
   let c3: Cpp.Complete = Cpp.foo(c1 + ({} as Cpp.Incomplete));
 }
 
+// ============================================================================
+// Incomplete operand Carbon type
+// ============================================================================
+
+// --- complete.h
+
+struct Complete {};
+
+// --- fail_incomplete_operand_carbon_type.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "complete.h";
+
+class Incomplete;
+fn CreateIncomplete() -> Incomplete*;
+
+fn F() {
+  var complete: Cpp.Complete = Cpp.Complete.Complete();
+  // CHECK:STDERR: fail_incomplete_operand_carbon_type.carbon:[[@LINE+14]]:21: error: looking up a C++ operator with incomplete operand type `Incomplete` [IncompletOperandTypeInCppOperatorLookup]
+  // CHECK:STDERR:   let result: i32 = *CreateIncomplete() + complete;
+  // CHECK:STDERR:                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  // CHECK:STDERR: fail_incomplete_operand_carbon_type.carbon:[[@LINE-8]]:1: note: class was forward declared here [ClassForwardDeclaredHere]
+  // CHECK:STDERR: class Incomplete;
+  // CHECK:STDERR: ^~~~~~~~~~~~~~~~~
+  // CHECK:STDERR: fail_incomplete_operand_carbon_type.carbon:[[@LINE+8]]:21: note: in `Cpp` operator `AddWith` lookup [InCppOperatorLookup]
+  // CHECK:STDERR:   let result: i32 = *CreateIncomplete() + complete;
+  // CHECK:STDERR:                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  // CHECK:STDERR:
+  // CHECK:STDERR: fail_incomplete_operand_carbon_type.carbon:[[@LINE+4]]:21: error: cannot access member of interface `Core.AddWith(Cpp.Complete)` in type `Incomplete` that does not implement that interface [MissingImplInMemberAccess]
+  // CHECK:STDERR:   let result: i32 = *CreateIncomplete() + complete;
+  // CHECK:STDERR:                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  // CHECK:STDERR:
```

Comment:
We'd like to only produce one error, something seems to not be propagating or looking for ErrorInst and producing another error in its presence. Let's fix or leave a TODO